__tests__/app.test.js

```diff
@@ -370,12 +378,59 @@ describe("GET /api/users", () => {
         });
       });
   });
-  test("", () => {});
-  test("", () => {});
-  test("", () => {});
-  test("", () => {});
-  test("", () => {});
-  test("", () => {});
+});
+
+describe("GET /api/articles?:filteringquery=:value", () => {
+  test("Status 200: responds with an array of articles filtered by topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(12);
+        articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("Status 200: responds with an array of articles filtered by author", () => {
+    return request(app)
+      .get("/api/articles?author=rogersop")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(3);
+        articles.forEach((article) => {
+          expect(article.author).toBe("rogersop");
+        });
+      });
+  });
+  test("Status 200: responds with an array of articles filtered by both topic and author", () => {
+    return request(app)
+      .get("/api/articles?author=rogersop&topic=mitch")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(2);
+        articles.forEach((article) => {
+          expect(article.author).toBe("rogersop");
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("Status 400: responds with an appropriate error when query key is invalid", () => {
+    return request(app)
+      .get("/api/articles?topix=mitch")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+  test("Status 404: responds with an appropriate error when query value does not produce any article", () => {
+    return request(app)
+      .get("/api/articles?topic=73")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
+  });
```

Comment:
Very good 👍 
Just one more test needed: Please make a request for a topic which exists (200) but that has no articles. This should return an empty array.